### PR TITLE
[special] alias for mvlgamma

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #50345 

Have added `out` variant for consistency.

TODO:
* [x] Check docs https://docs-preview.pytorch.org/61633/special.html#torch.special.multigammaln

### CODE REVIEW

**Comment by kshitij12345** [comment]
Gentle Ping @mruberry :)

File: aten/src/ATen/native/UnaryOps.cpp [hunk 1]
```diff
@@ -725,6 +725,34 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
   return self.copy_(args.lgamma_().sum(-1).add_(p2_sub_p * std::log(c10::pi<double>) * QUARTER));
 }
 
+Tensor& mvlgamma_out(const Tensor& self, int64_t p, Tensor& result) {
+  TORCH_CHECK(
```
Comment (mruberry) [inline aten/src/ATen/native/UnaryOps.cpp:729]: These device checks are now autogenerated (unless opted out of in native_functions.yaml), so I don't think we need this check anymore

**Review by mruberry — APPROVED** [review]
Nice work as usual, @kshitij12345

File: aten/src/ATen/native/UnaryOps.cpp (hunk 1 above)
Comment (kshitij12345) [reply aten/src/ATen/native/UnaryOps.cpp]: Thanks!

**Pull request closed without merging**